### PR TITLE
HIRO Media: Remove batching mechanism and use AJAX instead of JSONP

### PR DESCRIPTION
**Pull request by ronenst**
## Type of change
- [x] Refactoring (no functional changes, no api changes)

## Description of change
- Switched to AJAX instead of JSONP
- The mechanism that batched multiple requests according to specific keys has been removed. Each placement will now invoke a request.

These changes are related since the generated `batchKey` was also used as a request id in the JSONP responses. No such id is necessary with AJAX.

- contact email of the adapter’s maintainer
support@hiro-media.com

- [x] official adapter submission

### CODE REVIEW

File: src/adapters/hiromedia.js [hunk 1]
```diff
@@ -366,61 +278,72 @@ var HiroMediaAdapter = function HiroMediaAdapter() {
...
+            placementCode: bid.placementCode,
+            selectedSize: bidInfo.selectedSize,
+            additionalSizes: bidInfo.additionalSizes
+          });
 
-      // Loop over processed bids and send a request if a request for the bid
-      // batchKey has not been sent.
-      _bidStorage.forEach(function (bidInfo) {
+          Object.keys(requestParams).forEach(function (key){
+            if (requestParams[key] === '' || requestParams[key] === undefined) {
```
Comment (dbemiller) [inline src/adapters/hiromedia.js]: Same here.

**Comment by ronenst** [comment]
Thank you for the review, I've pushed a fix for the `undefined` issue.

You should be able test the adapter with the test account:
```
{
	bidder: 'hiromedia',
	params: {
		accountId: '1'
	}
}
```

**Comment by dbemiller** [comment]
Not seeing the ad when I test with those params...

```
<html>
<head>
    <!-- Prebid Config Section START -->
    <!-- Make sure this is inserted before your GPT tag -->
    <script>
      var PREBID_TIMEOUT = 700;

      var adUnits = [{
        code: 'div-gpt-ad-1460505748561-0',
        sizes: [[300, 250], [300,600]],
        bids: [{
          bidder: 'hiromedia',
          params: {
            accountId: '1'
          }
        }]
      }];

      var pbjs = pbjs || {};
      pbjs.que = pbjs.que || [];

    </script>
    <!-- Prebid Config Section END -->

    <!-- Prebid Boilerplate Section START. No Need to Edit. -->
    <script type="text/javascript" src="../build/dev/prebid.js" async></script>
    <script>
      var googletag = googletag || {};
      googletag.cmd = googletag.cmd || [];
      googletag.cmd.push(function() {
        googletag.pubads().disableInitialLoad();
      });

      pbjs.que.push(function() {
        pbjs.addAdUnits(adUnits);
        pbjs.requestBids({
          bidsBackHandler: sendAdserverRequest
        });
      });

      function sendAdserverRequest() {
        if (pbjs.adserverRequestSent) return;
        pbjs.adserverRequestSent = true;
        googletag.cmd.push(function() {
          pbjs.que.push(function() {
            pbjs.setTargetingForGPTAsync();
            googletag.pubads().refresh();
          });
        });
      }

      setTimeout(function() {
        sendAdserverRequest();
      }, PREBID_TIMEOUT);

    </script>
    <!-- Prebid Boilerplate Section END -->

    <script>
      (function () {
        var gads = document.createElement('script');
        gads.async = true;
        gads.type = 'text/javascript';
        var useSSL = 'https:' == document.location.protocol;
        gads.src = (useSSL ? 'https:' : 'http:') +
          '//www.googletagservices.com/tag/js/gpt.js';
        var node = document.getElementsByTagName('script')[0];
        node.parentNode.insertBefore(gads, node);
      })();
    </script>

    <script>
      googletag.cmd.push(function () {
        googletag.defineSlot('/19968336/header-bid-tag-0', [[300, 250], [300, 600]], 'div-gpt-ad-1460505748561-0').addService(googletag.pubads());

        googletag.pubads().enableSingleRequest();
        googletag.enableServices();
      });
    </script>
</head>

<body>
<h2>Prebid.js Test</h2>
<h5>Div-1</h5>
<div id='div-gpt-ad-1460505748561-0'>
    <script type='text/javascript'>
      googletag.cmd.push(function() { googletag.display('div-gpt-ad-1460505748561-0'); });
    </script>
</div>
</body>
</html>
```

**Comment by ronenst** [comment]
Please try again, it should work now, sorry about that.

**Comment by dbemiller** [comment]
Still no?

I see your server's response... but it's not making it back into the page.

screen shot 2017-05-04 at 9 11 29 am

**Comment by dbemiller** [comment]
I think because your bid doesn't have a cpm? Here's a `console.log(pbjs._bidsReceived)`

screen shot 2017-05-04 at 9 39 01 am

**Comment by ronenst** [comment]
Thanks again, should be resolved now. It was a Geo-location issue, I'm non-US so I was getting a different response. We'll make sure we have a stable test account for future testing.

**Comment by dbemiller** [comment]
gotcha... no worries. This LGTM now.

**Comment by dbemiller** [comment]
hey @ronenst... really sorry about the delay merging this, but it looks like some file conflicts popped up. Would you mind taking a look?

We're planning to release tomorrow... so I'll try to get another reviewer to take a look at it to get it in there.

Thanks

**Comment by ronenst** [comment]
Hi @dbemiller, thanks for the heads up, I've pushed a fix. Hopefully, we can manage to get it into the next release.

Please note the test account parameters have changed and the following parameters should be used:
```
{
  bidder: 'hiromedia',
  params: {
	accountId: '4'
  }
}
```

Since there are no code changes in the merge (just white-space), I've opted for a merge commit instead of a rebase so that the reviewed commit is comparable to the new commit, but the diff turned out strange on the GH page, I recommend checking it [with white-space changes ignored](https://github.com/prebid/Prebid.js/pull/1133/commits/89b84d028da13dbb5b102bc39ba64fb3796c65a6?w=1).

**Comment by dbemiller** [comment]
No problem at all. Still looks fine to me. Thanks!

**Pull request merged**